odoo/addons/base/models/ir_model.py

```diff
@@ -597,7 +597,10 @@ def _in_modules(self):
     @api.constrains('domain')
     def _check_domain(self):
         for field in self:
-            safe_eval(field.domain or '[]')
+            try:
+                safe_eval(field.domain or '[]')
+            except ValueError:
```

Comment:
Actually, `save_eval` can crash for many other reasons than a `ValueError`. By example, if someone enter `5 +`, you will get a `SyntaxError`, but it should also catch this kind of `Exception`.
```suggestion
            except Exception:
```

Also, the message is ambiguous, you don't check the validity of the domain but only the Python code execution.

Note that it doesn't make any sense to have use `safe_eval` here. Using `literal_eval` should be enough. I will create a small PR in master for that.

Reply:
Hello @ryv-odoo 

Changes are done.
Please look at it.
Thank you!